### ISSUE TITLE
sgrep can't find deep into multiple curly braces or subexpressions

### ISSUE THREAD

**Issue by DrewDennison** [returntocorp/sgrep#229]
**Describe the bug**
What if I want to find usage of a certain variable inside the same scope but deep into multiple curly braces (closures or if\else or maybe something else) without specifying it?

**To Reproduce**
See https://sgrep.live/pZJ
Copied here for preservation:

sgrep pattern:
```sgrep
var $X = "...";
...
function $F() {
    ...
    console.log($X);
}
```

JavaScript test case:
```
var foo = "bar"

function check() {
    if (foo === 'bar') {
        console.log(foo)   
    }
}
```

Error: is that there are no results

**Comment by nbrahms** [comment]
@aryx Is it possible to expand this to include try/except clauses?

Please see https://sgrep.live/JjZ for an example.

**Comment by ievans** [comment]
Also, it seems that we are not recursing into classes or functions in Javascript, see example:

https://sgrep.live/L3J

**Comment by dlukeomalley** [comment]
It appears this was shipped in 0.4.9? May I close the ticket?

**Comment by nbrahms** [comment]
I'm pretty sure matching into sub-statements works, but sub-expressions does not. So the ticket as written is incomplete. However, it probably makes sense to open a new ticket for expressions. Also, that may instead be satisfied using metavariable consistency from returntocorp/pfff#76.
